### PR TITLE
feat(travis): Test on Mac and Linux

### PR DESCRIPTION
**Pull request by codydaig**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,10 @@ node_js:
   - 0.12
   - 4
   - 5
+os:
+  - osx
+  - linux
```
Comment (lirantal) [inline .travis.yml]: any idea what linux varies from centos?

**Comment by lirantal** [comment]
Very cool @codydaig.
I re-ran the builds again as they failed only on Mac, maybe due to networking issues, let's see.

File: .travis.yml (hunk 1 above)
Comment (codydaig) [inline .travis.yml]: Nope. The online example shows osx and linux. So I just tried centos to see if that worked. I know we've been having a couple issues with centos on the generator side of things so I specifically want centos. I may swap out linux for ubuntu and see how that works. I want to play with these settings for a bit before we merge. Once we merge this in here, I'll also add them to the generator project.

**Comment by codydaig** [comment]
@lirantal Since when did we add CircleCI to the integration tests?

**Comment by lirantal** [comment]
@codydaig I never added Circle CI nor codeship.
Maybe @ilanbiala ?

**Comment by codydaig** [comment]
@lirantal Put it to just linux and centos. As long as these tests pass, and it looks good to you, I'll squash and merge. Then I'll work adding windows and mac. (Windows and mac both failed). I'll investigate what it will take to add those in. 

Codeship is the code coverage and has been there for a while, but CircleCI is new.

**Comment by lirantal** [comment]
Not so sure on Codeship. I enabled code coverage based on coveralls.io.
I think someone had some updates there.

**Comment by ilanbiala** [comment]
The reason it failed on Mac is because nvm isn't installed. I think Travis suggests installing using `brew`, but I'm not sure how we can do OS-specific stuff for pre build.

**Comment by ilanbiala** [comment]
@lirantal @codydaig I think the other CIs was a glitch, but can you add:

```
  - windows
  - macosx
```

**Comment by ilanbiala** [comment]
LGTM. @codydaig merge?

**Comment by lirantal** [comment]
Very cool guys.

**Pull request merged**